contracts/utils/BaalVotes.sol

```diff
@@ -0,0 +1,235 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "@openzeppelin/contracts/token/ERC20/extensions/draft-ERC20Permit.sol";
+
+/**
+ * @dev similar to Openzeplin ERC20Votes
+ * 
+ * uses timestamp instead of block.number and auto self delegates.
+ *
+ * This extension keeps a history (checkpoints) of each account's vote power. Vote power can be delegated either
+ * by calling the {delegate} function directly, or by providing a signature to be used with {delegateBySig}. Voting
+ * power can be queried through the public accessors  {getPriorVotes}.
+ *
+ * TODO alighn more with OZ ERC20Votes
```

Comment:
left over comment

Reply:
this is actually a thing i would like to do but probably out of scope for now maybe should remove it to keep it clean